phoenix-core/src/main/java/org/apache/phoenix/schema/types/FileSystemBasedLobStore.java

```diff
@@ -21,41 +21,62 @@
 import com.google.common.annotations.VisibleForTesting;
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang.RandomStringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.FileSystem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.io.FileInputStream;
-import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
 import java.io.InputStream;
 
 /**
  * Test Implementation to unblock others writes to /tmp/
  */
 public class FileSystemBasedLobStore implements LobStore {
 
-    private static final String ROOT_DIRECTORY = "/tmp/";
+    public static final Logger LOGGER =
+            LoggerFactory.getLogger(FileSystemBasedLobStore.class);
+    private static final Path ROOT_DIRECTORY = new Path("/user/hdfs/lob/hdfs/");
 
+    private FileSystem initializeHDFS() throws IOException {
```

Comment:
This should be its own implementation rather than altering the FileSystemBasedLobStore implementation. Something like `HdfsBasedLobStore`

Reply:
+1